fix(parser): record fields should not include language keywords

german1608 opened the pull request:
# Summary

Fixes #1750 

# Solution

Split `reservedKeywords` from `reservedIdentifiers`


Currently there are failing tests because [standard says that `Some` is a reserved keyword](https://github.com/dhall-lang/dhall-lang/issues/1012). Meanwhile, I'll have this PR as a draft

german1608 commented:
I've created [this](https://github.com/dhall-lang/dhall-lang/pull/1013) PR to add tests for this on common repository

german1608 commented:
Actual footage of me with all the issues/prs references I've made last hour:

![](https://media.tenor.com/images/14ddbf0aa32e2283fb7059e084a505eb/tenor.gif)

Gabriel439 commented:
As @sjakobi noted in https://github.com/dhall-lang/dhall-lang/issues/1012#issuecomment-630650616 you will need to add an exception for `Some` for record labels.  Otherwise, this looks great to me!

german1608 commented:
I added a `anyLabelOrSome` to have our parser impl similar as official grammar specs.

There were other places where I had to use `anyLabelOrSome`, like union types and field projections.

I've had to ammend some test files because it was using invalid record field labels.

I've added tests for this in dhall-lang/dhall-lang#1013

Gabriel439 commented on dhall/tests/format/applicationMultilineA.dhall:
```diff
@@ -18,7 +18,7 @@
     Some
       [ aaaaaaaaaaaaaaaaaaaaaaaaaa, bbbbbbbbbbbbbbbbbbbb, cccccccccccccccccccccccc, dddddddddddddd ]
-, merge =
+, merge1 =
```
For future reference, another thing you can do to keep the original name is to quote the label in backticks:

```dhall
, `merge` =
```

... but this is fine, too
